# Løsningsforslag - monopol og produktdifferensiering

Kaffebønna er lokalisert i Tromsø med flere utsalgssteder. I første omgang antar vi at Kaffebønna er monopolist i sitt marked hvor kundene er horisontalt differensiert (den linære byen). 

Anta at Kaffebønna har 30 000 kunder (N=30 000) som er jevnt fordelt langs lokaliseringsområdet til utsalgsstedene, og at hver kunde kjøper en kopp kaffe per dag. 

Kostnaden ved å produsere en kopp kaffe er NOK 5, og bedriften har faste kostnader per utsalgssted på NOK 25 000.

Transportkostnadene for kundene er på NOK 50.

Hva er optimalt antall utsalgssteder for Kaffebønna i Tromsø?

In [1]:
import sympy as sp
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# Definerer symbol
n, N ,V, c, pi, F,t, Q,Z,P = sp.symbols('n N V c \pi F t Q Z P')

# Definerer profittfunksjonen
profitt = sp.Eq(pi, (V-(t/(2*n))-c)*N-n*F)

display(profitt)

Eq(\pi, -F*n + N*(V - c - t/(2*n)))

In [3]:
# Deriverer profittfunksjonen mhp. n
profitt_N = sp.diff(profitt.rhs, n)

# Setter den deriverte lik 0 (python trenger ikke sette lik 0) og løser for n
n_løsning = sp.solve(profitt_N, n)

# Vi får to løsninger den ene er negativ og den andre er positiv, så derfor velger vi den positive
display(n_løsning[1])

sqrt(2)*sqrt(N*t/F)/2

In [4]:
# Verdier for kaffebønna
Optimalt_utsalg_løsning = n_løsning[1].subs({F: 25000, t: 50, N: 30000})

display(round(Optimalt_utsalg_løsning,2))

5.48

Optimalt antall utsalgsteder for kaffebønna finner vi ved å definere profittfunksjonen, denne er gitt ved:

$$
\pi = - F n + N \left(V - c - \frac{t}{2 n}\right)
$$

hvor $F$ er de faste kostnadene per utsalgssted, $n$ er antall utsalgssteder, $N$ er antall kunder, $V$ er kundenes betalingsvilje, $c$ er kostnaden ved å produsere en kopp kaffe og $t$ er transportkostnadene for kundene.

Vi vet at kaffebønna har 30 000 kunder, transportkostnadene er NOK 50, kostnaden ved å produsere en kopp kaffe er NOK 5, og de faste kostnadene per utsalgssted er NOK 25 000.

Vi kan finne optimalt antall utsalgssteder ved å derivere profittfunksjonen med hensyn på antall utsalgssteder og sette den deriverte lik null:

$$
\frac{d\pi}{dn} = -\frac{t}{2n^2} N - F = 0
$$

Merk at kostnadene for en kopp kaffe ikke er med i den deriverte profittfunksjonen, og derfor ikke påvirker optimalt antall utsalgssteder.

Til slutt løser vi for $n$:

$$
n^* = \sqrt{\frac{tN}{2F}}
$$

Og kan sette inn tallverdier for $N$, $t$ og $F$:

$$
n^* = \sqrt{\frac{50 \times 30 000}{2 \times 25 000}} = 5.48
$$

Og da finner vi at optimalt antall utsalgsteder for Kaffebønna blir 5.48.